test/e2e/upgrade/upgrade_test.go

```diff
@@ -25,6 +26,11 @@ var _ = Describe("Upgrade", Label("upgrade"), func() {
 			Expect(e2eclient.ClientsEnabled).To(BeTrue(), "failed to create runtime-controller client")
 		}
 		initialized = true
+		operatorVersion, err := version.OfOperator(context.TODO(), e2eclient.Client)
+		Expect(err).NotTo(HaveOccurred())
+		if operatorVersion.String() < "4.18" {
```

Comment:
Is this string comparison always valid?
Why not using https://github.com/hashicorp/go-version or
https://pkg.go.dev/golang.org/x/mod/semver ?

Reply:
the comment is generally correct, but we already use a version package (`aquasecurity`'s) please let's try to use that before to consider new packages

Reply:
@ffromani We already imported [semver](https://pkg.go.dev/golang.org/x/mod/semver), and `operatorVersion` is using the same type. so I have changed the code to use `semver` aswell